### DISCUSSION TITLE
Access GPIO to add Temperature Probe

### DISCUSSION THREAD

**Discussion by mholzma** [David00/rpi-power-monitor#120] [General]
I currently have the Monitor Shield on my Raspberry Pi and have been using it to monitor the heater that I installed in my garage. That heater is on an old-school thermostat. I was wondering what I would need to do to expose a GPIO pin so that I could add a temperature probe to the Raspberry Pi. Do I just need to solder a couple of headers to the existing shield or would that not be a good idea?

Thanks and really appreciate the great product.

**Comment by David00** [comment]
Hey there, Mike! Great question. Yes, you can solder directly onto the power monitor PCB pins with only a few exceptions. Both the DIY kit and the factory assembled PCB use the same header configuration, which you can see [here](https://github.com/David00/rpi-power-monitor/wiki/Schematic) (relevant part in the screenshot below):

![image](https://github.com/David00/rpi-power-monitor/assets/17997328/03068ffc-c7be-4779-89e9-80f1e99c117e)

You can use any of the other GPIO pins **except** for GPIO 8, 9, 10 and 11, as these are dedicated to the power monitor board.

I personally like the DS18B20 temperature sensor due to its high accuracy and simple connection.  This guide goes over connecting one, and even has an example script you can use:
https://raspberrypi-guide.github.io/electronics/temperature-monitoring

If you wanted to do something with the temperature _and_ the power monitor data together - (perhaps something like turning on an electric heater if your home is net exporting power, and the temperature is below 70F, for example) - then I'd recommend creating a plugin for the power monitor.  

Plugins are one of the newer features I've added to the project and it provides the means to extend the functionality of your power monitor with custom code that integrates cleanly. I've even put out a [simple GPIO controller plugin](https://david00.github.io/rpi-power-monitor/docs/v0.3.0/plugin-list.html#1-gpio-controller-example-plugin) that changes the state of a GPIO pin based on a specific part of the power monitor's data.  However, if you're just looking to use your existing Pi to also monitor temperatures and don't need to tie it into the power monitor in any way, there's not really any need to make a plugin.

**Reply by mholzma** [reply]
Thanks for the quick response, I'll get over my hesitancy of messing with the pre-fabricated board and add the pin extension, I'll just set it up for temperature monitoring at this time but the plugin sounds like a fun experiment.
